### PR TITLE
57 confman dbus

### PR DESCRIPTION
**Pull request by jbbjarnason**

### CODE REVIEW

File: libs/confman/src/detail/config_dbus_client.cpp [hunk 1]
```diff
@@ -0,0 +1,62 @@
...
+#include <tfc/progbase.hpp>
+
+#include <fmt/format.h>
+#include <boost/asio.hpp>
+#include <sdbusplus/asio/connection.hpp>
+#include <sdbusplus/asio/object_server.hpp>
+
+namespace tfc::confman::detail {
+
+static std::string replace_all(std::string const& input, std::string_view what, std::string_view with) {
```
Comment (omarhogni) [inline libs/confman/src/detail/config_dbus_client.cpp:15]: Why force the caller to make a copy. If you take a & and modify input directly this function becomes more useful.

File: libs/confman/tests/confman_test.cpp [hunk 2]
```diff
@@ -36,115 +35,166 @@ struct glz::meta<storage> {
...
+// static std::string replace_all(std::string const& input, std::string_view what, std::string_view with) {
+//   std::size_t count{};
+//   std::string copy{ input };
+//   for (std::string::size_type pos{}; std::string::npos != (pos = copy.find(what.data(), pos, what.length()));
+//        pos += with.length(), ++count) {
+//     copy.replace(pos, what.length(), with.data(), with.length());
+//   }
+//   return copy;
+// }
+
```
Comment (omarhogni) [inline libs/confman/tests/confman_test.cpp:47]:
```suggestion
```

File: libs/confman/tests/confman_test.cpp [hunk 3]
```diff
@@ -36,115 +35,166 @@ struct glz::meta<storage> {
...
+  //              }
+  //            });
+  //      });
+  //
+  //      ctx.run_for(std::chrono::milliseconds(200));
+  //    };
+  //  };
+  //  if (run_slow) {
+  //    ut::cfg<ut::override> = { .tag = { "slow" } };
+  //  }
```
Comment (omarhogni) [inline libs/confman/tests/confman_test.cpp:198]: Should we not just remove this?

File: libs/confman/src/detail/config_dbus_client.cpp (hunk 1 above)
Comment (omarhogni) [reply libs/confman/src/detail/config_dbus_client.cpp:15]: I see why you did this.

**Review by omarhogni — APPROVED** [review]

File: libs/confman/inc/public/tfc/confman/observable.hpp [hunk 4]
```diff
@@ -145,6 +146,17 @@ class [[nodiscard]] observable {
...
+// struct from_json<tfc::confman::observable<value_t>> {
+//   template <auto opts>
+//   inline static void op(auto& value, auto&&... args) noexcept {
+//     value_t value_copy;
+//     read<json>::op<opts>(value_copy, args...);
+////    from_json<value_t>::template op<opts>(value_copy, std::forward<decltype(args)>(args)...);
+//    value.set(std::move(value_copy)); // invoke callback
+//  }
+//};
+
```
[the remark this answers is not in the document]
Comment (jbbjarnason) [reply libs/confman/inc/public/tfc/confman/observable.hpp:159]: this is wip

File: libs/confman/tests/confman_test.cpp (hunk 2 above)
Comment (jbbjarnason) [reply libs/confman/tests/confman_test.cpp:47]: wip code

File: libs/confman/tests/confman_test.cpp (hunk 3 above)
Comment (jbbjarnason) [reply libs/confman/tests/confman_test.cpp:198]: wip code

**Pull request merged**